patches/0083-lavfi-dnn-Add-OpenVINO-API-2.0-support.patch

```diff
@@ -0,0 +1,1224 @@
++ * License along with FFmpeg; if not, write to the Free Software
++ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
++ */
++
++/**
++ * @file
++ * DNN OpenVINO backend implementation.
++ */
++
++#include "dnn_backend_common.h"
++#include "dnn_backend_openvino2.h"
++#include "dnn_io_proc.h"
++#include "libavutil/avassert.h"
++#include "libavutil/avstring.h"
++#include "libavutil/detection_bbox.h"
++#include "libavutil/cpu.h"
++#include "libavutil/opt.h"
++#include "safe_queue.h"
++#include "../internal.h"
++#include <openvino/c/openvino.h>
++
++typedef struct OV2Options {
++    char *device_type;
++    int nireq;
++    uint8_t async;
++    int batch_size;
++    int input_resizable;
++} OV2Options;
++
++typedef struct OV2Context {
++    const AVClass *class;
++    OV2Options options;
++} OV2Context;
++
++typedef struct OV2Model {
++    OV2Context ctx;
++    DNNModel *model;
++    ov_core_t *core;
++    ov_model_t *ov_model;
++    ov_compiled_model_t *compiled_model;
++    ov_output_const_port_t* input_port;
++    ov_preprocess_input_info_t* input_info;
++    ov_output_const_port_t* output_port;
++    ov_preprocess_output_info_t* output_info;
++    ov_preprocess_prepostprocessor_t* preprocess;
++
++    SafeQueue *request_queue;
++    Queue *lltask_queue;
++    Queue *task_queue;
++} OV2Model;
++
++typedef struct OV2RequestItem {
++    ov_infer_request_t *infer_request;
++    LastLevelTaskItem **lltasks;
++    uint32_t lltask_count;
++    ov_callback_t callback;
++} OV2RequestItem;
++
++#define OFFSET(x) offsetof(OV2Context, x)
++#define FLAGS AV_OPT_FLAG_FILTERING_PARAM
++static const AVOption dnn_openvino2_options[] = {
++    { "device", "device to run model", OFFSET(options.device_type), AV_OPT_TYPE_STRING, { .str = "CPU" }, 0, 0, FLAGS },
++    DNN_BACKEND_COMMON_OPTIONS
++    { "batch_size",  "batch size per request", OFFSET(options.batch_size),  AV_OPT_TYPE_INT,    { .i64 = 1 },     1, 1000, FLAGS},
++    { "input_resizable", "can input be resizable or not", OFFSET(options.input_resizable), AV_OPT_TYPE_BOOL,   { .i64 = 0 },     0, 1, FLAGS },
++    { NULL }
++};
++
++AVFILTER_DEFINE_CLASS(dnn_openvino2);
++
++static DNNDataType port_datatype_to_dnn_datatype(ov_element_type_e port_datatype)
++{
++    switch (port_datatype)
++    {
++    case F32:
++        return DNN_FLOAT;
++    case U8:
++        return DNN_UINT8;
++    default:
++        av_assert0(!"Not supported yet.");
++        return DNN_FLOAT;
++    }
++}
++
++static int fill_model_input_ov2(OV2Model *ov2_model, OV2RequestItem *request);
++static void infer_completion_callback(void *args);
++
++static int execute_model_ov2(OV2RequestItem *request, Queue *inferenceq)
++{
++    ov_status_e status;
++    LastLevelTaskItem *lltask;
++    int ret = 0;
++    TaskItem *task;
++    OV2Context *ctx;
++    OV2Model *ov2_model;
++
++    if (ff_queue_size(inferenceq) == 0) {
++        ov_infer_request_free(request->infer_request);
++        av_freep(&request);
```

Comment:
is there a same place to free such resources ?

Reply:
Yes, it is also freed in the end of callback func `infer_completion_callback`.
And other resources free would be in `ff_dnn_free_model_ov2`.